### PR TITLE
Homeandgamenewverion

### PR DESCRIPTION
**Pull request by ismail2009**
### Fixes
fixes #<issue_number>

### Related
related to #62 

### Summary
build signin and sign up page with his routers and authentication 

### Major Changes
[**OPTIONAL for small PRs**] Detailed description of major changes in the PR, ideally with references to relevant commits, issues, line-numbers, etc.

### CODE REVIEW

File: database/queries/user.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
+const db = require('../dbconnection');
+
+const addUser = user => new Promise((resolve, reject) => {
+  console.log('sdsds');
```
Comment (mhmdshorafa) [inline database/queries/user.js]: remove console log

Comment (ismail2009) [reply database/queries/user.js]: I solve it @mhmdshorafa

File: src/controllers/signup.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  res.render('signup', {
+    login: false,
+  });
+};
+
+exports.post = (req, res) => {
+  const { username, email, password } = req.body;
+  bcrypto.hashPassword(password).then((hashedPassword) => {
+    user.addUser({ username, email, password: hashedPassword })
+      .then((data) => {
```
Comment (mhmdshorafa) [inline src/controllers/signup.js:15]: you can do 
```javascript
if(data.rowCount) {
const { id, username } = data.rows[0];
const userData = {id, username};
```

File: src/controllers/signup.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+        const userData = {
+          id: data.rows[0].id,
+          username: data.rows[0].username,
+        };
+        const token = jwt.sign(userData, process.env.TOKEN_SECRET);
+        res.cookie('token', token);
+        res.json({ success: true, username: userData.username });
+      })
+      .catch(err => console.log(err));
```
Comment (mhmdshorafa) [inline src/controllers/signup.js]: please remove the console and return an action instead

File: src/controllers/signup.js [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+  }).catch((err) => { console.log(err); });
```
Comment (mhmdshorafa) [inline src/controllers/signup.js]: please remove the console and return an action instead

File: src/controllers/signin.js [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+    loginstyle: true,
+    login: false,
+    activePage: {
+      game: true,
+    },
+  });
+};
+exports.post = (req, res) => {
+  const { username, password } = req.body;
+  user.checkUser(username).then((data) => {
```
Comment (mhmdshorafa) [inline src/controllers/signin.js:17]: check if the data rows contains values ,, or data.rows[0] will cause a bug

File: src/controllers/home.js [hunk 6]
```diff
@@ -1,3 +1,15 @@
+const jwt = require('jsonwebtoken');
+require('env2')('config.env');
+
 exports.get = (req, res) => {
-  res.render('home', { home: true });
+  const cookie = req.cookies.token || '';
+  if (cookie) {
+    jwt.verify(cookie, process.env.TOKEN_SECRET, (error, d) => {
```
Comment (mhmdshorafa) [inline src/controllers/home.js]: rename d to something meaningfull

File: src/controllers/game.js [hunk 7]
```diff
@@ -1,6 +1,15 @@
+const jwt = require('jsonwebtoken');
+require('env2')('config.env');
 
 exports.get = (req, res) => {
-  res.render('game', {
-    game: true,
-  });
+  const cookie = req.cookies.token || '';
+  if (cookie) {
+    jwt.verify(cookie, process.env.TOKEN_SECRET, (error, d) => {
```
Comment (mhmdshorafa) [inline src/controllers/game.js]: name d to something with meaning

File: public/js/signupDom.js [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+        body: JSON.stringify(user),
+      }).then(res => res.json())
+        .then((data) => {
+          if (data.success) {
+            window.location.pathname = '/';
+          }
+        })
+        .catch(err => console.log(err));
+    } else {
+      alert('correct your password');
```
Comment (mhmdshorafa) [inline public/js/signupDom.js]: using alert is not a user friendly ,, please make sure to show a nice message

File: public/js/signupDom.js [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+        },
+        credentials: 'include',
+        body: JSON.stringify(user),
+      }).then(res => res.json())
+        .then((data) => {
+          if (data.success) {
+            window.location.pathname = '/';
+          }
+        })
+        .catch(err => console.log(err));
```
Comment (mhmdshorafa) [inline public/js/signupDom.js]: please take an action instead of logging the error

File: public/js/signupDom.js [hunk 10]
```diff
@@ -0,0 +1,34 @@
...
+    const email = document.getElementById('email');
+    const password = document.getElementById('password');
+    const cfPassword = document.getElementById('cf_password');
+    if (password.value === cfPassword.value) {
+      const user = {
+        username: username.value,
+        email: email.value,
+        password: username.value,
+      };
+      console.log(user);
```
Comment (mhmdshorafa) [inline public/js/signupDom.js]: remove log

File: public/js/signinDom.js [hunk 11]
```diff
@@ -0,0 +1,27 @@
...
+      credentials: 'include',
+      body: JSON.stringify(user),
+    })
+      .then(res => res.json())
+      .then((data) => {
+        if (data.success) {
+          window.location.pathname = '/game';
+        }
+      })
+      .catch(err => console.log(err));
```
Comment (mhmdshorafa) [inline public/js/signinDom.js:25]: please take an action instead of logging the error

**Review by mhmdshorafa — CHANGES_REQUESTED** [review]
good job guys ,, we need to change these things before we merge

**Pull request merged**